src/Atmos/Model/AtmosModel.jl

```diff
@@ -821,13 +824,22 @@ function source!(
 )
     ρu_pad = SVector(1, 1, 1)
     tend = Source()
-    ts = recover_thermo_state(m, state, aux)
-    args = (m, state, aux, t, ts, direction, diffusive)
-    source.ρ = Σsources(eq_tends(Mass(), m, tend), args...)
-    source.ρu = Σsources(eq_tends(Momentum(), m, tend), args...) .* ρu_pad
-    source.ρe = Σsources(eq_tends(Energy(), m, tend), args...)
-    source!(m.moisture, source, args...)
-    source!(m.precipitation, source, args...)
+
+    args = (
+        state = state,
+        aux = aux,
+        t = t,
+        direction = direction,
+        diffusive = diffusive,
+    )
+    precomputed = precompute(m, args, tend)
```

Comment:
we could do
```suggestion
    _args = (
        state = state,
        aux = aux,
        t = t,
        direction = direction,
        diffusive = diffusive,
    )
    args = merge(_args, (precomputed = precompute(m, args, tend),))
```
then not need to pass `precomputed` seperately.